Add parsl.addresses to reference guide

benclifford opened the pull request:
Partly addresses #648

yadudoc commented on parsl/addresses.py:
```diff
@@ -9,36 +9,52 @@
+    """Finds an address for the local host by querying the local routing table
+       for the route to Google DNS.
+
+       This will return an incorrect value when the internet-facing address is
```
This may not return a correct value when the network doesn't give an externally accessible address ?

yadudoc commented on parsl/addresses.py:
```diff
@@ -9,36 +9,52 @@
+    """Returns the hostname of the local host.
+
+       This will return an incorrect value when the hostname cannot be
+       resolved correctly from workers.
```
It is returning a correct value, but a value that may not be resolvable on the workers.

benclifford commented:
I think both your comments relate you having a different interpretation of what "correct" means... I mean correct as in "it will work when passed into htex as the public IP address" rather than "an address was successfully discovered using the described mechanism". Let me pick a better wording to reflect this.

yadudoc commented:
Fixes #648

benclifford commented:
I've changed the term '(in)correct' to '(un)usable' which I think better reflects what I intended.

I've also added a docstring for the whole module.
